### ISSUE TITLE
Is it possible to make Jackett simpler?

### ISSUE THREAD

**Issue by Omegaplexx** [Jackett/Jackett#15673]
### Is there already a request for your feature?

- [X] I have checked older issues, open and closed

### Is your feature request related to a problem? Please describe.

Hello, dear devs! Let me ask a question.

I have a potato laptop and Jackett runs slowly on it. Jack takes about 170 MB and contains more than 900 files when unarchived.

[TorrServer](https://github.com/YouROK/TorrServer), which streams torrents over HTTP, consists of a single compiled file and takes about 40 MB. It has binaries for a lot of platforms and architectures and runs fast on my slow computer.

I was wondering if Jackett could be made into something similar? Its purpose doesn't look any more complicated than TorrServer's. Could Jackett, for example, be rid of a number of heavy frameworks that limit its work on different platforms? Would it then be possible to port it to Android? Is there a ways to make it more simple, lightweight and portable?

### Describe the solution you'd like

I am not a programmer, so I'm sorry if there's something I don't understand about Jackett.

### Describe alternatives you've considered

I've been watching this repository for about a year and I'm amazed at your responsiveness and tireless work for such a long time. Thank you so much. I love Jackett!

**Comment by garfield69** [comment]
TLDR; probably, but it is unlikely to ever happen

The difference between the two apps, from what I can tell by a cursory look at the support site, is that to use TorrServer you first need to already have a magnet or torrent for it to do stuff with. Seems more akin to a download tool, similar as qBitTorrent, uTorrent, Transmission, Deluge, etc.

While Jackett is a tool that lets you search for torrents/magnets. The reason its is complex and has many files, is because it has a customised file for every one of the torrent tracker web sites that Jackett can talk to, Currently 572 of them, each one describing what each torrent tracker web site options are for searching for titles, what information each torrent tracker presents for the titles that are found, and how that information is is structured (HTML, JSON, XML) etc.
 
The rest of the differences are too technical to get into, suffice to say, that while Jackett's code could probably be optimised and cleaned up, there is no appetite from any of the C# devs to rewrite the app, its just too much work.
